public/app/core/utils/fetch.ts

```diff
@@ -93,7 +93,7 @@ export const parseBody = (options: BackendSrvRequest, isAppJson: boolean) => {
     return options.data;
   }
 
-  return isAppJson ? JSON.stringify(options.data) : new URLSearchParams(options.data);
+  return isAppJson ? safeStringifyValue(options.data) : new URLSearchParams(options.data);
```

Comment:
I feel a bit uneasy about doing this on every request, especially with a method that silently swallows errors...

Would it be possible for higher up in the stack to handle this? It's not clear to me why a circular-refed object is now being POSTed with scenes?

Reply:
The request includes scopedVars, which includes `__sceneObject`, which can contain circular references

Reply:
Someone (a plugin?) is sending the whole `scopedVars` in a request? 🥴 So I'm guessing we can't change how it's serialised?

Does `__sceneObject` need to be in scoped vars for dashboards? Could it be de-circular refed before data sources get it?

I fear that changing JSON.serialise for every request body to just silently send an empty string on invalid input is unintuitive, leads to more hard-to-spot bugs, and ultimately more harmful. The more this sits with me, the stronger I object to it 😅

Reply:
If the issue you have is that it fails silently, we could just not have the `try-catch` block in safeStringifyValue. Would that work for you?

Reply:
Nice - It would be better. I still think it's problematic to do this in the core fetch function, but I would be happy if someone else thought it was okay to break our tie 👍 

I think my main problem is that a circular reference usually indicates some other problem, the exception on circular reference is helpful to tell you that you're possibly sending something unexpected.

I think it could be useful to have this as a utility for consumers to opt into using, but I don't think it's appropriate to do this automatically in grafana's core fetch that everyone uses.

Reply:
@kaydelaney I understand Josh concern. We're removing circular dependencies in the payload to fail silently.

I'd remove the circular dependencies right before passing the payload to the backend service, in the dashboard code. So we don't apply this change to all the backend requests from the UI. I can be dangerous. WDYT?

Reply:
@ivanortegaalba I don't think there should be any concern if we remove the try-catch block in `safeStringifyValue` so that it only removes circular references when stringifying an object, but can still otherwise throw, considering that the alternative is for an object with circular references to get passed to JSON.stringify, causing a crash.
Further, I think trying to just remove circular references earlier in the backend, without immediately stringifying, could be quite tricky re: typing.

Reply:
> Does __sceneObject need to be in scoped vars for dashboards? Could it be de-circular refed before data sources get it?

@joshhunt  - I'm afraid it can not. Data sources often interpolate variables in their `query` method implementation and for that to happen in the Scenes arch, the entire object is necessary. Moreover, we need that object also to resolve a data source that's sourced from a data source variable (ref `getDataSourceSrv().get` calls). So every code path that can end up with variables replacement, this var is needed.

Reply:
I am ok with this. @joshhunt @dprokop any objections?

Reply:
I do object, but not strongly enough to block it 👍